src/orchestrator-process/src/lib.rs

```diff
@@ -525,6 +528,25 @@ impl NamespacedProcessOrchestrator {
                 }
             }
 
+            // Clean up scratch directory when the service is terminated.
+            // This is best effort as a development and testing convenience.
+            // Because the process orchestrator is not used in production, we
+            // don't need to be perfectly robust with the cleanup.
+            let _guard = scopeguard::guard((), |_| {
+                if let Some(scratch) = scratch_directory {
+                    info!(scratch_dir = %scratch.display(), "cleaning up scratch directory");
+                    task::spawn(
+                        || "clean_cluster_scratch_directory",
+                        remove_dir_all(scratch).map_err(|error| {
+                            warn!(
+                                "Error cleaning up scratch directory: {}",
+                                error.display_with_causes()
+                            )
+                        }),
```

Comment:
Ah, so the best thing to do here is to just introduce a _new_ future via an `async` block. Then you can await the `remove_dir_all` call so you can use `if let`, like you were before...

```suggestion
                        async {
                            if let Err(e) = remove_dir_all(scratch).await {
                                warn!(
                                    "Error cleaning up scratch directory: {}",
                                    error.display_with_causes()
                                );
                            }
                        }),
```

(It's more idiomatic to use `if let` if you're not interested in the output of the `map_err`.)

Reply:
I see what you mean, since we are not doing anything with the mapped result.

(Similar to preferring looping over an iterator vs mapping it if we are not particularly returning anything)

Reply:
Updated